From 065f4425199d355a9cfda17c8fe884fd897875c4 Mon Sep 17 00:00:00 2001
From: Harry Wentland <harry.wentland@amd.com>
Date: Tue, 4 Oct 2016 16:37:59 -0400
Subject: [PATCH 1217/1722] drm/amd/dal: Don't use power module for virtual
 sinks

There were a couple ASSERTs getting hit because we called
the power module for virtual sinks. It makes no sense to use
the power module for virtual sinks.

Change-Id: I37d493ed3228190067d209f49b64b0cba86656d2
Signed-off-by: Harry Wentland <harry.wentland@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/modules/power/power.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/drivers/gpu/drm/amd/dal/modules/power/power.c b/drivers/gpu/drm/amd/dal/modules/power/power.c
index fbc666c..ea07e84 100644
--- a/drivers/gpu/drm/amd/dal/modules/power/power.c
+++ b/drivers/gpu/drm/amd/dal/modules/power/power.c
@@ -230,6 +230,9 @@ void mod_power_destroy(struct mod_power *mod_power)
 bool mod_power_add_sink(struct mod_power *mod_power,
 						const struct dc_sink *sink)
 {
+	if (sink->sink_signal == SIGNAL_TYPE_VIRTUAL)
+		return false;
+
 	struct core_power *core_power =
 				MOD_POWER_TO_CORE(mod_power);
 	struct core_dc *core_dc = DC_TO_CORE(core_power->dc);
@@ -288,6 +291,13 @@ bool mod_power_set_backlight(struct mod_power *mod_power,
 	union dmcu_abm_set_bl_params params;
 
 	for (stream_index = 0; stream_index < num_streams; stream_index++) {
+		if (streams[stream_index]->sink->sink_signal == SIGNAL_TYPE_VIRTUAL) {
+			core_power->state[sink_index].backlight = 0;
+			core_power->state[sink_index].frame_ramp = 0;
+			core_power->state[sink_index].smooth_brightness_enabled = false;
+			continue;
+		}
+
 		sink_index = sink_index_from_sink(core_power,
 				streams[stream_index]->sink);
 
@@ -322,6 +332,9 @@ bool mod_power_get_backlight(struct mod_power *mod_power,
 		const struct dc_sink *sink,
 		unsigned int *backlight_8bit)
 {
+	if (sink->sink_signal == SIGNAL_TYPE_VIRTUAL)
+		return false;
+
 	struct core_power *core_power =
 				MOD_POWER_TO_CORE(mod_power);
 
@@ -586,6 +599,9 @@ bool mod_power_get_panel_backlight_boundaries(
 bool mod_power_set_smooth_brightness(struct mod_power *mod_power,
 		const struct dc_sink *sink, bool enable_brightness)
 {
+	if (sink->sink_signal == SIGNAL_TYPE_VIRTUAL)
+		return false;
+
 	struct core_power *core_power =
 			MOD_POWER_TO_CORE(mod_power);
 	unsigned int sink_index = sink_index_from_sink(core_power, sink);
@@ -598,6 +614,9 @@ bool mod_power_set_smooth_brightness(struct mod_power *mod_power,
 bool mod_power_notify_mode_change(struct mod_power *mod_power,
 		const struct dc_stream *stream)
 {
+	if (stream->sink->sink_signal == SIGNAL_TYPE_VIRTUAL)
+		return false;
+
 	struct core_power *core_power =
 			MOD_POWER_TO_CORE(mod_power);
 
-- 
2.7.4

